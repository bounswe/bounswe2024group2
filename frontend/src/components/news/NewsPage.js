import React, { useEffect, useState } from 'react';
import NewsCard from './NewsCard';
import FilterButtons from './FilterButtons';
import '../../styles/news/News.css';

const mockNewsData = [
    {
        id: 1,
        title: "GDP Growth Forecasts Revised",
        description: "Recent reports indicate that GDP growth forecasts for major economies have been revised upwards.",
        rssUrl: "https://example.com/news1",
        category: "Global Economy",
        publishedAt: "2 hours ago", 
        coverImageUrl: "https://www.thisiscolossal.com/wp-content/uploads/2024/10/jonk-1.jpg", 
        source: "Economic Times",
    },
    {
        id: 2,
        title: "Major Indices Reach All-Time Highs",
        description: "The stock market sees a significant uptick as major indices reach all-time highs.",
        rssUrl: "https://example.com/news2",
        category: "Stock Market",
        publishedAt: "3 hours ago",
        // coverImageUrl: "https://example.com/images/news2.jpg", 
        source: "Market Watch", 
    },
    {
        id: 3,
        title: "Bitcoin Surges Past $60,000",
        description: "Bitcoin's value has surged past $60,000 amid increased institutional adoption.",
        rssUrl: "https://example.com/news3",
        category: "Cryptocurrency",
        publishedAt: "1 hour ago", 
        // coverImageUrl: "https://example.com/images/news3.jpg", 
        source: "CoinDesk", 
    },
    {
        id: 4,
        title: "Rising Costs Impact Consumers",
        description: "Inflation rates are rising, leading to increased costs for consumers on everyday goods.",
        rssUrl: "https://example.com/news4",
        category: "Inflation",
        publishedAt: "4 hours ago", 
        // coverImageUrl: "https://example.com/images/news4.jpg", 
        source: "Bloomberg", 
    },
    {
        id: 5,
        title: "Unemployment Rates Hit Record Low",
        description: "The job market strengthens as unemployment rates hit a record low across the nation.",
        rssUrl: "https://example.com/news5",
        category: "Job Market",
        publishedAt: "5 hours ago", 
        coverImageUrl: "https://1finance.co.in/magazine/wp-content/uploads/2024/01/Blog-5-b-scaled.jpg", 
        source: "The Wall Street Journal",
    },
];

// Mock data for RSS feeds
const mockRssFeedData = [
    {
        id: 6,
        title: "New Agreements Boost International Trade",
        description: "New trade agreements are set to boost international trade among participating countries.",
        rssUrl: "https://example.com/news6",
        category: "Trade",
        publishedAt: "3 days ago",
        // coverImageUrl: "https://example.com/images/news6.jpg",
        source: "Reuters",
    },
    {
        id: 7,
        title: "Emerging Markets on the Rise",
        description: "Emerging markets are gaining traction as investors look for new opportunities.",
        rssUrl: "https://example.com/news7",
        category: "Market Trends",
        publishedAt: "2 days ago", 
        // coverImageUrl: "https://example.com/images/news7.jpg",
        source: "Forbes", 
    },
    {
        id: 8,
        title: "Interest Rates Held Steady",
        description: "The Federal Reserve has announced that interest rates will remain steady for the foreseeable future.",
        rssUrl: "https://example.com/news8",
        category: "Federal Reserve",
        publishedAt: "1 day ago", 
        // coverImageUrl: "https://example.com/images/news8.jpg",
        source: "CNBC", 
    },
];


const NewsPage = () => {
    const [newsData, setNewsData] = useState([]);
    const [rssNewsData, setRssNewsData] = useState([]);
    const all = 'All';
    const [selectedCategory, setSelectedCategory] = useState(all);
    const [selectedRssCategory, setSelectedRssCategory] = useState(all);

    useEffect(() => {
        setNewsData(mockNewsData);
        setRssNewsData(mockRssFeedData);
    }, []);

    const newsCategories = [all, ...new Set(mockNewsData.map(news => news.category))];
    const rssCategories = [all, ...new Set(mockRssFeedData.map(news => news.category))];

    const filteredNews = newsData.filter((news) =>
        selectedCategory === all ? true : news.category === selectedCategory
    );

    const filteredRssNews = rssNewsData.filter((news) =>
        selectedRssCategory === all ? true : news.category === selectedRssCategory
    );

    return (
        <div className="news-page">
            <div className="news-header">
                <h1 className="news-title">Economic News</h1>
                <h2 className="news-subtitle">Your daily updates on economic trends</h2>
            </div>

            <div className="news-content">
                <h3 className="news-section-title">Mock Economic News Section</h3>
                <div className="news-filter-buttons">
                    <FilterButtons categories={newsCategories} setSelectedCategory={setSelectedCategory} selectedCategory={selectedCategory} />    
                </div>
                <div className="news-scrollable-section">
                    {filteredNews.map((news) => (
                        <NewsCard key={news.id} news={news} />
                    ))}
                </div>

                <h3 className="news-section-title">Mock RSS Feed Economic News Section</h3>
                <div className="news-filter-buttons">
                    <FilterButtons categories={rssCategories} setSelectedCategory={setSelectedRssCategory} selectedCategory={selectedRssCategory} />
                </div>
                <div className="news-scrollable-section">
                    {filteredRssNews.map((news) => (
                        <NewsCard key={news.id} news={news} />
                    ))}
                </div>
            </div>
        </div>
    );
};

export default NewsPage;
