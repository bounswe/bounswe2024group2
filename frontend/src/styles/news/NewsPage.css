.filter-buttons {
    display: flex;
    gap: 10px;
    justify-content: flex-start;
    margin-bottom: 20px;
}

.filter-buttons .filter-button {
    background-color: var(--color-primary-500);
    border: none;
    border-radius: 5px;
    color: var(--color-neutral-white);
    cursor: pointer;
    padding: 8px 12px;
    transition: background-color 0.3s ease;
}

.filter-buttons .filter-button.selected {
    font-weight: bold;
    background-color: var(--color-primary-700);
}

.filter-buttons .filter-button:hover {
    background-color: var(--color-primary-700);
}


.news-scrollable-section {
    display: grid;
    gap: 32px;
    grid-template-columns: repeat(3, 1fr);
    margin-bottom: 20px;
}

.news-section-title {
    color: var(--color-neutral-800);
    font-size: 2rem;
    margin: 20px 0;
    text-align: left;
}

@media (max-width: 768px) {
    .news-scrollable-section {
        grid-template-columns: repeat(2, 1fr);
    }
}

@media (max-width: 480px) {
    .news-scrollable-section {
        grid-template-columns: 1fr;
    }
}
