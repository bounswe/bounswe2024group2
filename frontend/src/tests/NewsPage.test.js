import React from 'react';
import { render } from '@testing-library/react';
import NewsPage from '../components/news/NewsPage';

test('renders Login correctly', () => {
  const { asFragment } = render(<NewsPage title="Snapshot Testing" />);
  
  expect(asFragment()).toMatchSnapshot();
});