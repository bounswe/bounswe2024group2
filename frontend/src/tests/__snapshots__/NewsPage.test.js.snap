// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login correctly 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="page-header"
    >
      <h1
        class="page-title"
      >
        Economic News
      </h1>
      <h2
        class="page-subtitle"
      >
        Your daily updates on economic trends
      </h2>
    </div>
    <div
      class="page-content"
    >
      <h3
        class="news-section-title"
      >
        Mock Economic News Section
      </h3>
      <div
        class="news-filter-buttons"
      >
        <div
          class="filter-buttons"
        >
          <button
            class="filter-button selected"
          >
            All
          </button>
          <button
            class="filter-button "
          >
            Global Economy
          </button>
          <button
            class="filter-button "
          >
            Stock Market
          </button>
          <button
            class="filter-button "
          >
            Cryptocurrency
          </button>
          <button
            class="filter-button "
          >
            Inflation
          </button>
          <button
            class="filter-button "
          >
            Job Market
          </button>
        </div>
      </div>
      <div
        class="news-scrollable-section"
      >
        <div
          class="news-card"
        >
          <div
            class="news-card-header"
          >
            <span
              class="news-card-source"
            >
              Economic Times
            </span>
            <span
              class="news-card-time"
            >
              2 hours ago
            </span>
          </div>
          <div
            class="news-card-body"
          >
            <div
              class="news-card-text"
            >
              <h3
                class="news-card-title"
              >
                GDP Growth Forecasts Revised
              </h3>
              <p
                class="news-card-description"
              >
                Recent reports indicate that GDP growth forecasts for major economies have been revised upwards.
              </p>
            </div>
            <div
              class="news-card-icon"
            >
              <img
                alt="Economic Times"
                src="https://www.thisiscolossal.com/wp-content/uploads/2024/10/jonk-1.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="news-card"
        >
          <div
            class="news-card-header"
          >
            <span
              class="news-card-source"
            >
              Market Watch
            </span>
            <span
              class="news-card-time"
            >
              3 hours ago
            </span>
          </div>
          <div
            class="news-card-body"
          >
            <div
              class="news-card-text"
            >
              <h3
                class="news-card-title"
              >
                Major Indices Reach All-Time Highs
              </h3>
              <p
                class="news-card-description"
              >
                The stock market sees a significant uptick as major indices reach all-time highs.
              </p>
            </div>
          </div>
        </div>
        <div
          class="news-card"
        >
          <div
            class="news-card-header"
          >
            <span
              class="news-card-source"
            >
              CoinDesk
            </span>
            <span
              class="news-card-time"
            >
              1 hour ago
            </span>
          </div>
          <div
            class="news-card-body"
          >
            <div
              class="news-card-text"
            >
              <h3
                class="news-card-title"
              >
                Bitcoin Surges Past $60,000
              </h3>
              <p
                class="news-card-description"
              >
                Bitcoin's value has surged past $60,000 amid increased institutional adoption.
              </p>
            </div>
          </div>
        </div>
        <div
          class="news-card"
        >
          <div
            class="news-card-header"
          >
            <span
              class="news-card-source"
            >
              Bloomberg
            </span>
            <span
              class="news-card-time"
            >
              4 hours ago
            </span>
          </div>
          <div
            class="news-card-body"
          >
            <div
              class="news-card-text"
            >
              <h3
                class="news-card-title"
              >
                Rising Costs Impact Consumers
              </h3>
              <p
                class="news-card-description"
              >
                Inflation rates are rising, leading to increased costs for consumers on everyday goods.
              </p>
            </div>
          </div>
        </div>
        <div
          class="news-card"
        >
          <div
            class="news-card-header"
          >
            <span
              class="news-card-source"
            >
              The Wall Street Journal
            </span>
            <span
              class="news-card-time"
            >
              5 hours ago
            </span>
          </div>
          <div
            class="news-card-body"
          >
            <div
              class="news-card-text"
            >
              <h3
                class="news-card-title"
              >
                Unemployment Rates Hit Record Low
              </h3>
              <p
                class="news-card-description"
              >
                The job market strengthens as unemployment rates hit a record low across the nation.
              </p>
            </div>
            <div
              class="news-card-icon"
            >
              <img
                alt="The Wall Street Journal"
                src="https://1finance.co.in/magazine/wp-content/uploads/2024/01/Blog-5-b-scaled.jpg"
              />
            </div>
          </div>
        </div>
      </div>
      <h3
        class="news-section-title"
      >
        Mock RSS Feed Economic News Section
      </h3>
      <div
        class="news-filter-buttons"
      >
        <div
          class="filter-buttons"
        >
          <button
            class="filter-button selected"
          >
            All
          </button>
          <button
            class="filter-button "
          >
            Trade
          </button>
          <button
            class="filter-button "
          >
            Market Trends
          </button>
          <button
            class="filter-button "
          >
            Federal Reserve
          </button>
        </div>
      </div>
      <div
        class="news-scrollable-section"
      >
        <div
          class="news-card"
        >
          <div
            class="news-card-header"
          >
            <span
              class="news-card-source"
            >
              Reuters
            </span>
            <span
              class="news-card-time"
            >
              3 days ago
            </span>
          </div>
          <div
            class="news-card-body"
          >
            <div
              class="news-card-text"
            >
              <h3
                class="news-card-title"
              >
                New Agreements Boost International Trade
              </h3>
              <p
                class="news-card-description"
              >
                New trade agreements are set to boost international trade among participating countries.
              </p>
            </div>
          </div>
        </div>
        <div
          class="news-card"
        >
          <div
            class="news-card-header"
          >
            <span
              class="news-card-source"
            >
              Forbes
            </span>
            <span
              class="news-card-time"
            >
              2 days ago
            </span>
          </div>
          <div
            class="news-card-body"
          >
            <div
              class="news-card-text"
            >
              <h3
                class="news-card-title"
              >
                Emerging Markets on the Rise
              </h3>
              <p
                class="news-card-description"
              >
                Emerging markets are gaining traction as investors look for new opportunities.
              </p>
            </div>
          </div>
        </div>
        <div
          class="news-card"
        >
          <div
            class="news-card-header"
          >
            <span
              class="news-card-source"
            >
              CNBC
            </span>
            <span
              class="news-card-time"
            >
              1 day ago
            </span>
          </div>
          <div
            class="news-card-body"
          >
            <div
              class="news-card-text"
            >
              <h3
                class="news-card-title"
              >
                Interest Rates Held Steady
              </h3>
              <p
                class="news-card-description"
              >
                The Federal Reserve has announced that interest rates will remain steady for the foreseeable future.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
